src/apps/natdiscovery/natdiscovery.c

```diff
@@ -55,7 +55,7 @@ static int counter = 0;
 
 #ifdef __cplusplus
 
-static int init_socket(int *socketfd, ioa_addr *local_addr, int local_port, ioa_addr *remote_addr) {
+static int init_socket(int *socketfd, ioa_addr *local_addr, uint16_t local_port, ioa_addr *remote_addr) {
   int ret = 0;
 
   if (local_port >= 0) {
```

Comment:
This conditional is now always true.

Make sure you go through the entire data flow for each port variable you are changing to confirm you're properly handling error conditions.

Reply:
You are right. I have reverted the change